utils/plugin/plugin.go

```diff
@@ -0,0 +1,125 @@
+package plugin
```

Comment:
Is this file just used for tests? If so, it is better to move to the /test folder.